.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,67 @@
+name: Build with emscripten
+
+on:
+  push:
+    branches:
+      - "master"
+      - "flake"
+      - "ga"
+  pull_request:
+    branches:
+      - "master"
+
+  # run workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install cached emscripten
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: 'emsdk'
+          key: 'emsdk'
+
+      - name: Install emscripten
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          curl -L https://github.com/emscripten-core/emsdk/archive/refs/tags/3.1.30.tar.gz | tar xz
+          mv emsdk-* emsdk
+          ./emsdk/emsdk install latest
+          ./emsdk/emsdk activate latest
+
+      - name: Build blink
+        run: |
+          . ./emsdk/emsdk_env.sh
+          emmake make -j$(nproc)
+          mkdir o//blink/blink-web
+          cp third_party/coi-serviceworker/coi-serviceworker.js o//blink/blink-web/coi-serviceworker.js
+          emcc -O2 o//blink/blink.o o//blink/blink.a -lm -pthread -lrt -o o//blink/blink-web/blink.html \
+            --shell-file blink/blink-shell.html \
+            -s INVOKE_RUN=0 \
+            -s INITIAL_MEMORY=1073741824 \
+            -s EXIT_RUNTIME=1 \
+            -s EXPORTED_RUNTIME_METHODS=callMain \
+            -fno-builtin-exit \
+            -sASSERTIONS
+
+      - name: Deploy sample to GitHub pages
+        uses: JamesIves/github-pages-deploy-action@v4.4.1
```

Comment:
For 3rd party actions I usually recommend to use the full commit hash to prevent that the tag is retroactively changed. Especially here there is write access to the repo possible.

Reply:
Done.